### DISCUSSION TITLE
Railroad Signals

### DISCUSSION THREAD

**Discussion by ConstanceHorne** [OpenTTD/OpenTTD#8494] [Features]
I am a player that originally started with Transport Tycoon, then TTD and stopped after Windows 95 left the building. I even used TTD Patch for a time. I will say I only just discovered OpenTTD in the last couple of weeks.

I cannot wrap my mind around the new railroad signaling. I have watched every video I can find on Youtube and read everything I can find anywhere and I still don't really understand the signaling in Open TTD. 

I don't think I can be the only person like this. I am also a developer (for the last 40 years) so I am also in the process of setting up to download the source and look at it myself. 

I believe the signaling has become incomprehensible, since TTD. I understand the need for options but there is no documentation showing how to use the signals, at least not in any depth at all. Even the wording is not self-explanatory. 

I am willing to update the wiki to provide instructions -- if I only understood how they worked. I am very good at documentation for end users, in fact. I get tapped to do that at work all the time and I love doing it. <I know, I'm weird.>

Also the tutorial game doesn't even cover railroads _at_all_. It's a TBD (to be done) item. Fortunately, I know how to build rail but the signals, WOW, that's a tough nut to crack for me right now.

I saw on IRC how you all are chatting. I am listening and trying to learn but need some help here.

Constance

**Comment by kabal2020** [comment]
Have you come accross the wiki page; https://wiki.openttd.org/en/Manual/Signals?

For majority of cases, just use the path signals/one way path signals, these are the signals that have been added most recently if I recall.  

The others are older, and still have some specific use cases but tend to be required for more for advances play now.

**Comment by ConstanceHorne** [comment]
Yes, of course.

**Comment by andythenorth** [comment]
Hi Constance :)

The signals are confusing in this order:
* original TTD signals are very limited, but left in place for compatibility reasons.
  * they could be _ignored_ by everybody except those opening an old savegame
* the pre-signals are very powerful, but very complicated, and most players will never need or want to use them for most ways of playing the game
  * they could be _ignored_ by most people, except those building complex, self-regulating networks
* path-based signals (PBS) are reliable and easy to use, but have a few quirks
  * they are what _most_ players need most of the time, and what most _new_ players should use by default

How to communicate that remains a puzzle. 😃

**Reply by Andrew350** [reply]
Semi-serious suggestion: Add a new setting under Interface -> Construction -> _Enable advanced signals_ and leave it off by default, showing only path signals in the GUI initially.

It won't help players figure out _how_ to use them, but makes the choice of _what_ to use much easier.

**Reply by frosch123** [reply]
I found some old note on my disk:
1. Replace "Cycle through signal types" enum setting with a bool setting "Cycle block and path signals separately".
2. Make Ctrl on build no longer toggle semaphore, but instead toggle between "default signal type" and block/oneway-path depending on whether the default is block or path.
3. Add a GUI setting: Build "only path", "both", "only block" signals, which disables the signals in both the GUI and the cycling and ctrl thingies.
